diff --git a/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h b/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h
index 75ba8e143e40bce101f6b1b839e6accd5c0ed7b6..dffdba96f67e328eb95fdb6429b879e4eb840451 100644
--- a/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h
+++ b/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h
@@ -288,7 +288,7 @@ class ReactNativeFeatureFlagsDefaults : public ReactNativeFeatureFlagsProvider {
   }
 
   bool preventShadowTreeCommitExhaustion() override {
-    return false;
+    return true;
   }
 
   bool shouldPressibilityUseW3CPointerEventsForHover() override {
