diff --git a/Libraries/Image/Image.android.js b/Libraries/Image/Image.android.js
index f5e3f8a64ef948912080182a5bd86633f58faaf3..9680d7d1de7aaf4c92c2417c7a30129ee646dec8 100644
--- a/Libraries/Image/Image.android.js
+++ b/Libraries/Image/Image.android.js
@@ -124,6 +124,7 @@ const EMPTY_IMAGE_SOURCE = {
   uri: undefined,
   width: undefined,
   height: undefined,
+  headers: undefined,
 };
 
 /**
@@ -215,10 +216,18 @@ let BaseImage: AbstractImageAndroid = ({
     ];
     nativeProps.source = source_;
   } else {
-    const {uri, width: sourceWidth, height: sourceHeight} = source_;
+    const {
+      uri,
+      width: sourceWidth,
+      height: sourceHeight,
+      headers: sourceHeaders,
+    } = source_;
     if (uri === '') {
       console.warn('source.uri should not be an empty string');
     }
+    if (sourceHeaders != null) {
+      nativeProps.headers = sourceHeaders;
+    }
     nativeProps.style = [
       {width: sourceWidth ?? width, height: sourceHeight ?? height},
       styles.base,
